packages/sdk-core/src/BatchCall.ts

```diff
@@ -89,10 +92,25 @@ export default class BatchCall {
         const operationStructArray = await Promise.all(
             this.getOperationStructArrayPromises
         );
-        const tx =
-            this.host.contract.populateTransaction.batchCall(
-                operationStructArray
-            );
+
+        const values = operationStructArray
+            .filter((x) => x.value?.gt(BigNumber.from(0)))
+            .map((x) => x.value);
+
```

Comment:
A possible situation to protect against would be that the operation with the ETH value would be the first payable call in the set of operations.